sockeye/utils.py

```diff
@@ -142,6 +142,22 @@ def compute_lengths(sequence_data: mx.sym.Symbol) -> mx.sym.Symbol:
     return mx.sym.sum(sequence_data != C.PAD_ID, axis=1)
 
 
+def compute_length_ratio(input_seq: mx.nd.NDArray, label_seq: mx.nd.NDArray) -> mx.nd.NDArray:
+    """
+    Computes the ratio of label lengths to the input lengths.
+
+    :param input_seq: Input sequence. Shape: (batch_size, seq_len, 1).
+    :param label_seq: Label sequence. Shape: (batch_size, seq_len).
+    :return: Length data. Shape: (batch_size,).
+    """
+    # input_lengths: (batch_size, )
+    input_lengths = (input_seq != C.PAD_ID).sum(axis=1).squeeze()
```

Comment:
you can avoid the squeeze if you set keepdims=False in sum()

Reply:
I think it's False by default, the squeeze is there to remove the original 3rd axis.